Core/Src/spi_command_handler.c

```diff
@@ -5,170 +5,199 @@
+ *
+ * @param
+ * 		*tx_data: pointer to transmit data
+ * 		data_length: numbers of bytes to be sent
+ */
 void spi_transmit(uint8_t *tx_data, uint16_t data_length) {
-	HAL_SPI_Transmit(&hspi1, tx_data, data_length, HAL_MAX_DELAY);
+    HAL_SPI_Transmit(&hspi1, tx_data, data_length, HAL_MAX_DELAY);
 }
 
-void spi_receive(uint8_t *rx_data, uint16_t data_length) {
-	HAL_SPI_Receive(&hspi1, rx_data, data_length, HAL_MAX_DELAY);
+/**
+ * @brief
+ * 		Receive data of given size over SPI bus in interrupt mode
+ * @param
+ * 		*rx_data: pointer to receive data
+ * 		data_length: numbers of bytes to be receive
+ */
+void spi_receive(uint8_t *rx_data, uint16_t data_length) { HAL_SPI_Receive_IT(&hspi1, rx_data, data_length); }
+
+/**
+ * @brief
+ * 		Verifies if command from OBC is valid or not
+ * @param
+ * 		obc_cmd: Command from OBC
+ * @return
+ * 		1 if valid command, 0 if not
+ */
+int spi_verify_command(uint8_t obc_cmd) {
+    transmit_ack = 0;
+
+    switch (obc_cmd) {
+    case IRIS_SEND_HOUSEKEEPING: {
+        transmit_ack = 1;
+        break;
+    }
+    case IRIS_TAKE_PIC: {
+        transmit_ack = 1;
+        break;
+    }
+    case IRIS_GET_IMAGE_COUNT: {
+        transmit_ack = 1;
+        break;
+    }
+    case IRIS_TRANSFER_IMAGE: {
+        transmit_ack = 1;
+        break;
+    }
+    case IRIS_OFF_SENSOR_IDLE: {
+        transmit_ack = 1;
+        break;
+    }
+    case IRIS_ON_SENSOR_IDLE: {
+        transmit_ack = 1;
+        break;
+    }
+    case IRIS_GET_IMAGE_LENGTH: {
+        transmit_ack = 1;
+        break;
+    }
+    case IRIS_UPDATE_SENSOR_I2C_REG: {
+        transmit_ack = 1;
+        break;
+    }
+    case IRIS_UPDATE_CURRENT_LIMIT: {
+        transmit_ack = 1;
+        break;
+    }
+    }
+
+    if (transmit_ack) {
+        spi_transmit(&ack, 1);
+        return 0;
+    } else {
+        spi_transmit(&nack, 1);
+        return -1;
+    }
 }
 
 /**
- * @brief SPI command verifier
+ * @brief
+ * 		Handles command from OBC
+ * @param
+ * 		obc_cmd: Command from OBC
+ * @return
+ * 		1 if valid command, 0 if not
  */
-int spi_verify_command() {
-	uint8_t ack = 0xAA;
-	uint8_t nack = 0x0F;
-
-	spi_receive(&cmd, 1);
-
-	switch (cmd) {
-	case IRIS_SEND_HOUSEKEEPING: {
-		spi_transmit(&ack, 1);
-		return 0;
-	}
-	case IRIS_TAKE_PIC: {
-		spi_transmit(&ack, 1);
-		return 0;
-	}
-	case IRIS_GET_IMAGE_COUNT: {
-		spi_transmit(&ack, 1);
-		return 0;
-	}
-	case IRIS_TRANSFER_IMAGE: {
-		spi_transmit(&ack, 1);
-		return 0;
-	}
-	case IRIS_OFF_SENSOR_IDLE: {
-		spi_transmit(&ack, 1);
-		return 0;
-	}
-	case IRIS_ON_SENSOR_IDLE: {
-		spi_transmit(&ack, 1);
-		return 0;
-	}
-	case IRIS_GET_IMAGE_LENGTH: {
-		spi_transmit(&ack, 1);
-		return 0;
-	}
-	case IRIS_UPDATE_SENSOR_I2C_REG: {
-		spi_transmit(&ack, 1);
-		return 0;
-	}
-	case IRIS_UPDATE_CURRENT_LIMIT: {
-		spi_transmit(&ack, 1);
-		return 0;
-	}
-	default: {
-		spi_transmit(&nack, 1);
-		return -1;
-	}
-	}
+int spi_handle_command(uint8_t obc_cmd) {
```

Comment:
add wdt case so this doesn't return -1. I don't think it matters, but if and when we get into error handling we'll need it

Reply:
The current state machine will process any sort of `obc_cmd` by going into the `HANDLE_COMMAND` state. So yes, we will need to have the watchdog timer case